Find a functionally equivalent code:android.bluetooth.BluetoothSocket.read:COMMENT
Method Modifier: 
Comment:/*package*/

@@ -1,4 +1,6 @@
 {
+    if (mSocketIS == null)
+        throw new IOException("read is called on null InputStream");
     if (VDBG)
         Log.d(TAG, "read in:  " + mSocketIS + " len: " + length);
     int ret = mSocketIS.read(b, offset, length);

